packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -392,11 +391,10 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
     while (child != null) {
       final double childWidth = child.getMaxIntrinsicWidth(double.infinity);
       final double childHeight = child.getMaxIntrinsicHeight(childWidth);
-      if (runWidth + childWidth > width) {
-        height += runHeight;
-        if (runCount > 0)
-          height += runSpacing;
-        runCount += 1;
+      final double spaceWidth = childCount > 0 ? spacing : 0;
+      // child count check for situation when width of first child in run over available width
+      if (runWidth + childWidth + spaceWidth > width && childCount > 0) {
```

Comment:
nit: to parallel the code in layout change this to:

```suggestion
      if (childCount > 0 && runWidth + childWidth + spaceWidth > width) {
```